test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -8546,6 +8550,418 @@ public async Task RestoreNetCore_ProjectProvidedRuntimeIdentifierGraphChange_Doe
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                        pathContext.PackageSource,
+                        PackageSaveMode.Defaultv3,
+                        packages);
+                projectA.AddPackageToAllFrameworks(packages);
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                // Act
+                var r = Util.RestoreSolution(pathContext);
+
+                // Assert
+                r.Success.Should().BeTrue();
+                Assert.True(File.Exists(projectA.AssetsFileOutputPath));
+                Assert.True(File.Exists(projectA.NuGetLockFileOutputPath));
+
+                // Setup - change project's packages
+                projectA.CleanPackagesFromAllFrameworks();
+                packages = updatedPackageIdAndVersion.Select(p =>
+                {
+                    var id = p.Split('/')[0];
+                    var version = p.Split('/')[1];
+                    var package = new SimpleTestPackageContext()
+                    {
+                        Id = id,
+                        Version = version
+                    };
+                    package.Files.Clear();
+                    package.AddFile($"lib/netcoreapp2.0/{id}.dll");
+                    package.AddFile($"ref/netcoreapp2.0/{id}.dll");
+                    package.AddFile($"runtimes/win7/lib/netcoreapp2.0/{id}.dll");
+                    return package;
+                }).ToArray();
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                        pathContext.PackageSource,
+                        PackageSaveMode.Defaultv3,
+                        packages);
+                projectA.AddPackageToAllFrameworks(packages);
+                projectA.Save();
+
+                // Act
+                r = Util.RestoreSolution(pathContext, 1);
+
+                // Assert
+                r.Success.Should().BeFalse();
+                Assert.Contains("NU1004", r.Errors);
+            }
+        }
+
+        [Theory]
+        [InlineData(new string[] { "x_lockmodetdepch/1.0.0" }, new string[] { "x_lockmodetdepch/2.0.0" })]
+        [InlineData(new string[] { "x_lockmodetdepch/1.0.0" }, new string[] { "y_lockmodetdepch/1.0.0" })]
+        [InlineData(new string[] { "x_lockmodetdepch/1.0.0" }, new string[] { "x_lockmodetdepch/1.0.0", "y_lockmodetdepch/1.0.0" })]
+        [InlineData(new string[] { "x_lockmodetdepch/1.0.0", "y_lockmodetdepch/1.0.0" }, new string[] { "y_lockmodetdepch/1.0.0" })]
+        public async Task RestoreNetCore_PackagesLockFile_WithDependentProjectChangeOfPackageDependencyAndLockedMode_FailsRestore(
+           string[] initialPackageIdAndVersion,
+           string[] updatedPackageIdAndVersion)
+        {
+            // A project with RestoreLockedMode should fail restore if the package dependencies of a dependent project were changed between restores.
```

Comment:
Just FYI, we generally call this transient dependencies.

Reply:
transitive* :)

They mean different things :D

Reply:
I will update these comments in a future commit.